### PR TITLE
[JENKINS-58732] Remove docker-workflow dependency

### PR DESCRIPTION
**Pull request by chriskilding**
Continuation of JENKINS-58732

### CODE REVIEW

**Comment by jglick** [comment]
Some sort of `RequireUpperBoundDeps` error. Try using a 4.x POM.

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.57</version>
+    <version>4.1</version>
```
Comment (jglick) [inline pom.xml:29]:
```suggestion
    <version>4.2</version>
```

**Comment by jglick** [comment]
Try removing `access-modifier-annotation` from `pom.xml`.

**Comment by chriskilding** [comment]
That's moved it forward a bit, looks like just a hamcrest version conflict now (some things want 1.3, some things want 2.2).

**Comment by bitwiseman** [comment]
@chriskilding @jglick 
If you can resolve the merge for this, we can add this to the release I'll be publishing today.

**Comment by jglick** [comment]
I can try to fix the merge…

**Comment by jglick** [comment]
Resolved locally and could push if needed (assuming I have permission), but consider #393 instead.

**Pull request closed without merging**